bootloader/grub.go

```diff
@@ -180,27 +191,84 @@ func (g *grub) RemoveKernelAssets(s snap.PlaceInfo) error {
 func (g *grub) makeKernelEfiSymlink(s snap.PlaceInfo, name string) error {
 	// use a relative symlink destination so that it resolves properly, if grub
 	// is located at /run/mnt/ubuntu-boot or /boot/grub, etc.
-	extractedKernel := filepath.Join(
+	target := filepath.Join(
 		filepath.Base(s.MountFile()),
 		"kernel.efi",
 	)
 
+	// the location of the destination symlink as an absolute filepath
+	source := filepath.Join(g.dir(), name)
+
 	// check that the kernel snap has been extracted already so we don't
 	// inadvertently create a dangling symlink
 	// expand the relative symlink from g.dir()
-	if !osutil.FileExists(filepath.Join(g.dir(), extractedKernel)) {
+	if !osutil.FileExists(filepath.Join(g.dir(), target)) {
 		return fmt.Errorf(
 			"cannot enable %s at %s: %v",
 			name,
-			extractedKernel,
+			target,
 			os.ErrNotExist,
 		)
 	}
 
-	return os.Symlink(
-		extractedKernel,
-		filepath.Join(g.dir(), name),
-	)
+	// check if the target symlink already exists, if it doesn't then we can
```

Comment:
Perhaps it would be cleaner to create a reusable atomic symlink update function and use it here?

Reply:
Do we have other places we need to atomically update a symlink? Note that this function is already itself a helper, it's used both for `EnableKernel` (and the kernel.efi symlink) and `EnableTryKernel` (and the try-kernel.efi symlink).

Reply:
Hmm I think that, at least at this must-work level, it's only here. It's fine, I was just wondering.